sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -41,213 +43,303 @@
-        duration = time.time() - request.context["start_time"] if "start_time" in request.context else None
-        status_code = response.http_response.status_code
-        sub_status_str = response.http_response.headers.get("x-ms-substatus")
-        sub_status_code = int(sub_status_str) if sub_status_str else None
-        verb = request.http_request.method
-        http_version_obj = None
-        url = None
-        if self.diagnostics_handler:
+
+        if self._enable_diagnostics_logging:
+            context = request.context
+            http_response = response.http_response
+            headers = request.http_request.headers
+            sub_status_str = http_response.headers.get("x-ms-substatus")
+            sub_status_code: Optional[int] = int(sub_status_str) if sub_status_str else None
+            url_obj = http_response.internal_response.url  # type: ignore[attr-defined, union-attr]
             try:
-                major = response.http_response.internal_response.version.major  # type: ignore[attr-defined, union-attr]
-                minor = response.http_response.internal_response.version.minor  # type: ignore[attr-defined, union-attr]
-                http_version_obj = f"{major}."
-                http_version_obj += f"{minor}"
-            except (AttributeError, TypeError):
-                http_version_obj = None
+                duration = float(http_response.headers.get("x-ms-request-duration-ms"))  # type: ignore[union-attr]
+            except (ValueError, TypeError) as e:
+                duration = (time.time() - context["start_time"]) * 1000 if "start_time" in context else None
+
+            log_data = {"duration": duration,
+                        "status_code": http_response.status_code, "sub_status_code": sub_status_code,
+                        "verb": request.http_request.method,
+                        "operation_type": headers.get('x-ms-thinclient-proxy-operation-type'),
+                        "url": str(url_obj), "database_name": None, "collection_name": None,
+                        "resource_type": headers.get('x-ms-thinclient-proxy-resource-type'), "is_request": False}
+
+            if log_data["url"]:
+                url_parts: List[str] = log_data["url"].split('/')  # type: ignore[union-attr]
+                if 'dbs' in url_parts:
+                    dbs_index = url_parts.index('dbs')
+                    if dbs_index + 1 < len(url_parts):
+                        log_data["database_name"] = url_parts[dbs_index + 1]
+                if 'colls' in url_parts:
+                    colls_index = url_parts.index('colls')
+                    if colls_index + 1 < len(url_parts):
+                        log_data["collection_name"] = url_parts[colls_index + 1]
+
+            options = context.options
+            logger = context.setdefault("logger", options.pop("logger", self.logger))
+
+            context["logger_attributes"] = log_data.copy()
+            self.on_request(request)
```

Comment:
We're re-logging the request during the response?

Reply:
We are only logging the request when we get to response if the logger has any filters. Otherwise we wouldn't be able to filter related requests and responses together as the request wouldn't have attributes like duration or status code. If no filters are applied request are logged on requests.